bin/reth/src/chain/import.rs

```diff
@@ -0,0 +1,168 @@
+        verbatim_doc_comment,
+        default_value = "mainnet",
+        value_parser = genesis_value_parser
+    )]
+    chain: ChainSpec,
+
+    /// The path to a block file for import.
+    ///
+    /// The online stages (headers and bodies) are replaced by a file import, after which the
+    /// remaining stages are executed.
+    #[arg(long, value_name = "IMPORT_PATH", verbatim_doc_comment)]
+    path: PlatformPath<ConfigPath>,
+}
+
+impl ImportCommand {
+    /// Execute `import` command
+    pub async fn execute(self) -> eyre::Result<()> {
+        info!(target: "reth::cli", "reth {} starting", crate_version!());
+
+        let config: Config = self.load_config()?;
+        info!(target: "reth::cli", path = %self.db, "Configuration loaded");
+
+        info!(target: "reth::cli", path = %self.db, "Opening database");
+        let db = Arc::new(init_db(&self.db)?);
+        info!(target: "reth::cli", "Database opened");
+
+        debug!(target: "reth::cli", chainspec=?self.chain, "Initializing genesis");
+        init_genesis(db.clone(), self.chain.clone())?;
+
+        // create a new FileClient
+        info!(target: "reth::cli", "Importing chain file");
+        let file_client = Arc::new(FileClient::new(&self.path).await?);
+
+        // override the tip
+        let tip = file_client.tip().expect("file client has no tip");
+        info!(target: "reth::cli", "Chain file imported");
+
+        let (consensus, notifier) = BeaconConsensus::builder().build(self.chain.clone());
+        debug!(target: "reth::cli", %tip, "Tip manually set");
+        notifier.send(ForkchoiceState {
+            head_block_hash: tip,
+            safe_block_hash: tip,
+            finalized_block_hash: tip,
+        })?;
+        info!(target: "reth::cli", "Consensus engine initialized");
+
+        let (mut pipeline, events) =
+            self.build_import_pipeline(config, db.clone(), &consensus, file_client).await?;
+
+        tokio::spawn(handle_events(events));
+
+        // Run pipeline
+        info!(target: "reth::cli", "Starting sync pipeline");
+        tokio::select! {
+            res = pipeline.run(db.clone()) => res?,
+            _ = tokio::signal::ctrl_c() => {},
+        };
+
+        info!(target: "reth::cli", "Finishing up");
+        Ok(())
+    }
+
+    async fn build_import_pipeline<C>(
+        &self,
+        config: Config,
+        db: Arc<Env<WriteMap>>,
+        consensus: &Arc<C>,
+        file_client: Arc<FileClient>,
+    ) -> eyre::Result<(Pipeline<Env<WriteMap>, impl SyncStateUpdater>, impl Stream<Item = NodeEvent>)>
+    where
+        C: Consensus + 'static,
+    {
+        let header_downloader = ReverseHeadersDownloaderBuilder::from(config.stages.headers)
+            .build(consensus.clone(), file_client.clone())
+            .as_task();
+
+        let body_downloader = BodiesDownloaderBuilder::from(config.stages.bodies)
+            .build(file_client.clone(), consensus.clone(), db)
```

Comment:
Nit we should have consistent ordering in these arguments between downloaders..let's track in separate issue